From 480a64769e3c873f5f33434be857ce3c86844964 Mon Sep 17 00:00:00 2001
Message-Id: <480a64769e3c873f5f33434be857ce3c86844964.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Praveen Paneri <praveen.paneri@intel.com>
Date: Wed, 28 Jan 2015 15:40:13 +0530
Subject: [PATCH 08/95] SQUASHME! [VPG]: drm/i915: Generic pipe implementation
 for update DPST

Internal panels can use different pipe configuration.
This patch generalizes the DPST register handling according
to the active pipe for internal display for VLV.

The Orginal issue was introduced in:
	commit ca3e5be7439d2d92d31a75242f509ee5a4fcd789
	Author: sgurram <sudheer.gurram@intel.com>
	Date:   Mon Jan 20 19:00:09 2014 +0530
	MUST_REBASE [VPG]: drm/i915: Enable DPST on BYT

Issue: GMINL-5600
Change-Id: I43379f3924360e1a67e62386b1ace6f672f0cadf
Signed-off-by: sgurram <sudheer.gurram@intel.com>
Signed-off-by: Praveen Paneri <praveen.paneri@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index 7b5f5a3..a29f830 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -382,9 +382,12 @@ i915_dpst_update_registers(struct drm_device *dev)
 		dev_priv->dpst.reg.blm_hist_bin_count_mask = BIN_COUNT_MASK_4M;
 		dev_priv->dpst.ie_mod_table_enable = IE_MOD_TABLE_ENABLE;
 	} else if (IS_VALLEYVIEW(dev)) {
-		dev_priv->dpst.reg.blm_hist_ctl = VLV_BLC_HIST_CTL(PIPE_A);
-		dev_priv->dpst.reg.blm_hist_guard = VLV_BLC_HIST_GUARD(PIPE_A);
-		dev_priv->dpst.reg.blm_hist_bin = VLV_BLC_HIST_BIN(PIPE_A);
+		dev_priv->dpst.reg.blm_hist_ctl =
+				VLV_BLC_HIST_CTL(dev_priv->dpst.pipe);
+		dev_priv->dpst.reg.blm_hist_guard =
+				VLV_BLC_HIST_GUARD(dev_priv->dpst.pipe);
+		dev_priv->dpst.reg.blm_hist_bin =
+				VLV_BLC_HIST_BIN(dev_priv->dpst.pipe);
 		dev_priv->dpst.reg.blm_hist_bin_count_mask = BIN_COUNT_MASK_4M;
 		dev_priv->dpst.ie_mod_table_enable = VLV_IE_MOD_TABLE_ENABLE;
 	} else if (IS_BROADWELL(dev)) {
-- 
1.7.9.5

